src/tree/VirtualMachineTreeItem.ts

```
@@ -84,20 +86,38 @@ export class VirtualMachineTreeItem extends AzureTreeItem {
+            localize('Deleting', 'Deleting {0} "{1}"...', resourcesToDelete[0].resourceType, resourcesToDelete[0].resourceName);
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: `${deleting} Check the [output channel](command:${ext.prefix}.showOutputChannel) for status.` }, async (): Promise<void> => {
+            if (multiDelete) { ext.outputChannel.appendLog(deleting); }
+
+            const failedResources: string[] = await deleteAllResources(this.root, this.resourceGroup, resourcesToDelete);
+
+            const messageDeleteWithErrors: string = localize(
+                'messageDeleteWithErrors',
+                `Failed to delete the following resources ${failedResources.join(', ')}.`);
 
-        await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: deleting }, async (): Promise<void> => {
-            ext.outputChannel.appendLog(deleting);
-            await computeClient.virtualMachines.deleteMethod(this.resourceGroup, this.name);
-            vscode.window.showInformationMessage(deleteSucceeded);
-            ext.outputChannel.appendLog(deleteSucceeded);
+            const deleteSucceeded: string = multiDelete ? localize('DeleteSucceeded', 'Successfully deleted {0}.', context.resourceList) :
+                localize('DeleteSucceeded', 'Successfully deleted {0} "{1}".', resourcesToDelete[0].resourceType, resourcesToDelete[0].resourceName);
+
+            // single resources are already displayed in the output channel
+            if (multiDelete) { ext.outputChannel.appendLog(failedResources.length > 0 ? messageDeleteWithErrors : deleteSucceeded); }
+
+            if (failedResources.length > 0) {
+                context.errorHandling.suppressDisplay = true;
+                vscode.window.showErrorMessage(`${messageDeleteWithErrors} Check the [output channel](command:${ext.prefix}.showOutputChannel) for more information.`);
```

Pretty sure you could just throw an error with this message and it would work the exact same without `suppressDisplay` and `showErrorMessage` here. You only really need `showErrorMessage` here if you're doing your own buttons

---

Well, the problem I was running into was the errorHandler was automatically outputting the error message to the output channel because my error doesn't contain any new lines.